test/unit/es6-Template.js

```diff
@@ -0,0 +1,128 @@
+// This file uses es6 features and is loaded conditionally.
+
+var assert    = require('assert');
+var SqlString = require('../../');
+var test      = require('utest');
+
+var sql = SqlString.sql;
+
+function runTagTest (golden, test) {
+  // Run multiply to test memoization bugs.
+  for (let i = 3; --i >= 0;) {
+    let result = test();
+    if (typeof result.toSqlString !== 'string') {
+      result = result.toSqlString();
+    } else {
+      throw new Error(`Expected raw not ${result}`);
+    }
+    assert.equal(result, golden);
+  }
+}
+
+test('template tag', {
+  'numbers': function () {
+    runTagTest(
+      'SELECT 2',
+      () => sql`SELECT ${1 + 1}`);
+  },
+  'date': function () {
+    runTagTest(
+      `SELECT '2000-01-01 00:00:00.000'`,
+      () => sql({ timeZone: 'GMT' })`SELECT ${new Date(Date.UTC(2000, 0, 1, 0, 0, 0))}`);
+  },
+  'string': function () {
+    runTagTest(
+      `SELECT 'Hello, World!\\n'`,
+      () => sql`SELECT ${'Hello, World!\n'}`);
+  },
+  'identifier': function () {
+    runTagTest(
+      'SELECT `foo`',
+      () => sql`SELECT ${SqlString.identifier('foo')}`);
+  },
+  'blob': function () {
+    runTagTest(
+      'SELECT "\x1f8p\xbe\\\'OlI\xb3\xe3\\Z\x0cg(\x95\x7f"',
+      () =>
+        sql`SELECT "${Buffer.from("1f3870be274f6c49b3e31a0c6728957f","hex")}"`
```

Comment:
In case incremental diffs aren't working, I added this and the following two testcases to address other bugs that you found.